src/lib.rs

```diff
@@ -312,242 +312,216 @@ impl MapOptions {
+
 pub struct Handle {
-    on_resize: Closure<dyn Fn(JsValue)>,
-    on_remove: Closure<dyn Fn(JsValue)>,
-
-    // Interaction
-    on_mousedown: Closure<dyn Fn(JsValue)>,
-    on_mouseup: Closure<dyn Fn(JsValue)>,
-    on_preclick: Closure<dyn Fn(JsValue)>,
-    on_click: Closure<dyn Fn(JsValue)>,
-    on_dblclick: Closure<dyn Fn(JsValue)>,
-    on_mousemove: Closure<dyn Fn(JsValue)>,
-    on_mouseover: Closure<dyn Fn(JsValue)>,
-    on_mouseenter: Closure<dyn Fn(JsValue)>,
-    on_mouseleave: Closure<dyn Fn(JsValue)>,
-    on_mouseout: Closure<dyn Fn(JsValue)>,
-    on_contextmenu: Closure<dyn Fn(JsValue)>,
-    on_touchstart: Closure<dyn Fn(JsValue)>,
-    on_touchend: Closure<dyn Fn(JsValue)>,
-    on_touchcancel: Closure<dyn Fn(JsValue)>,
-    on_wheel: Closure<dyn Fn(JsValue)>,
-
-    // Movement
-    on_movestart: Closure<dyn Fn(JsValue)>,
-    on_move: Closure<dyn Fn(JsValue)>,
-    on_moveend: Closure<dyn Fn(JsValue)>,
-    on_dragstart: Closure<dyn Fn(JsValue)>,
-    on_drag: Closure<dyn Fn(JsValue)>,
-    on_dragend: Closure<dyn Fn(JsValue)>,
-    on_zoomstart: Closure<dyn Fn(JsValue)>,
-    on_zoom: Closure<dyn Fn(JsValue)>,
-    on_zoomend: Closure<dyn Fn(JsValue)>,
-    on_rotatestart: Closure<dyn Fn(JsValue)>,
-    on_rotate: Closure<dyn Fn(JsValue)>,
-    on_rotateend: Closure<dyn Fn(JsValue)>,
-    on_pitchstart: Closure<dyn Fn(JsValue)>,
-    on_pitch: Closure<dyn Fn(JsValue)>,
-    on_pitchend: Closure<dyn Fn(JsValue)>,
-    on_boxzoomstart: Closure<dyn Fn(JsValue)>,
-    on_boxzoomend: Closure<dyn Fn(JsValue)>,
-    on_boxzoomcancel: Closure<dyn Fn(JsValue)>,
-
-    // Lifecycle
-    on_load: Closure<dyn Fn(JsValue)>,
-    on_render: Closure<dyn Fn(JsValue)>,
-    on_idle: Closure<dyn Fn(JsValue)>,
+    $(
+        $event: Closure<dyn Fn(JsValue)>,
+    )*
     on_error: Closure<dyn Fn(JsValue)>,
-    on_webglcontextlost: Closure<dyn Fn(JsValue)>,
-    on_webglcontextrestored: Closure<dyn Fn(JsValue)>,
-
-    // Data loading
-    on_data: Closure<dyn Fn(JsValue)>,
-    on_styledata: Closure<dyn Fn(JsValue)>,
-    on_sourcedata: Closure<dyn Fn(JsValue)>,
-    on_dataloading: Closure<dyn Fn(JsValue)>,
-    on_styledataloading: Closure<dyn Fn(JsValue)>,
-    on_sourcedataloading: Closure<dyn Fn(JsValue)>,
-    on_styleimagemissing: Closure<dyn Fn(JsValue)>,
 }
 
-macro_rules! impl_handler{
-    ($(($event:ident, $type:ident),)*) => {
+    };
+}
+
+macro_rules! impl_handle {
+    ($(($event:ident, $type:ident);)*) => {
+
 impl Handle {
     pub fn new<F: MapEventListener + 'static>(map: Weak<Map>, f: F) -> Handle {
         let f = Rc::new(RefCell::new(f));
         Handle {
-            $($event: impl_event!(map, f, $event, $type),)*
-        }
-    }
-}
-    }
-}
-
-macro_rules! impl_event {
-    ($m: ident, $f:ident, $event:ident, JsValue) => {
-            Closure::new(enclose!(
-                ($m, $f) move |value: JsValue| {
-                    web_sys::console::debug_2(&JsValue::from(stringify!($event)), &value);
```

Comment:
Dropping this line in the new macro is the only actual diff by the way! Otherwise the new macros should generate exactly identical code to what existed before.. I'm not sure if it was important to have this debug statement but it made the console basically unusable the moment any handler is attached. Or maybe I did something wrong?